### PR TITLE
Path: support move method

### PR DESCRIPTION
**Pull request by stdweird**
Add support for `move`ing (/renaming) files and directories

### CODE REVIEW

**Comment by jrha** [comment]
LGTM, could do with a brief description

**Comment by stdweird** [comment]
@jrha is that sufficient?

File: src/main/perl/Path.pm [hunk 1]
```diff
@@ -147,7 +153,7 @@ sub _reset_exception_fail
     if ($EC->error()) {
         # LC::Exception supports formatted stringification
         my $errmsg = ''.$EC->error();
-        $self->debug(1, "Ignoring/resetting previous existing error: $errmsg");
+        $self->debug(1, "Ignoring/resetting previous existing error$msg: $errmsg");
```
Comment (jrha) [inline src/main/perl/Path.pm]: What does adding `$msg` here do? I've missed the context.

**Comment by jrha** [comment]
Yep, thanks.

File: src/main/perl/Path.pm (hunk 1 above)
Comment (stdweird) [inline src/main/perl/Path.pm]: it's an additional argument you can pass, mainly to get an idea what method is resetting any previously existing fail or exception. very usefull while debugging.

**Pull request merged**